### PR TITLE
Decompile windows and breaking tiles

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

#1692 #1883

### CODE REVIEW

**Comment by aredfan** [comment]
I found an issue where the masked goon is unable to enter Lara's bedroom from the balcony, in HSH.

1. Load the attached savegame: [savegame.zip](https://github.com/user-attachments/files/17757656/savegame.zip)
2. Have Lara jump onto the same tile as the shotgun
3. Then have Lara run back out into the bedroom and you'll see the masked goon is unable enter the bedroom through the broken windows. Then load the savegame in OG to compare.

File: src/tr2/game/objects/general/window.c [hunk 1]
```diff
@@ -18,7 +27,98 @@ void Window2_Setup(void)
...
+    item->mesh_bits = 1;
+
+    const ROOM *const r = Room_Get(item->room_num);
+    const int32_t z_sector = (item->pos.z - r->pos.z) >> WALL_SHIFT;
+    const int32_t x_sector = (item->pos.x - r->pos.x) >> WALL_SHIFT;
+    const SECTOR *const sector = &r->sectors[z_sector + x_sector * r->size.z];
+    BOX_INFO *const box = &g_Boxes[sector->box];
+
+    if (box->overlap_index & BOX_BLOCKABLE) {
+        box->overlap_index = BOX_BLOCKED;
```
Comment (lahm86) [inline src/tr2/game/objects/general/window.c]: Probably a typo - I think it's `|=`.

**Review by lahm86 — COMMENTED** [review]
LGTM, just the pathfinding issue. The start of Bartoli's has the same problem as HSH.

File: src/tr2/game/objects/general/window.c (hunk 1 above)
Comment (rr-) [reply src/tr2/game/objects/general/window.c]: Thanks – that was it. Fixed!

**Review by lahm86 — APPROVED** [review]

**Review by aredfan — APPROVED** [review]

**Review by walkawayy — APPROVED** [review]

**Pull request merged**